### PR TITLE
qemu_storage.QemuImg.rebase: support luks

### PR DESCRIPTION
**Pull request by lolyu**
1. add support of luks in rebase, which will cover:
1.1. rebase to a luks image
1.2. rebase a luks-inside-qcow2 image

Signed-off-by: lolyu <lolyu@redhat.com>

### CODE REVIEW

**Comment by lolyu** [comment]
```
(1/2) Host_RHEL.m8.u1.raw.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.rebase_onto_no_backing_file.compat_level.compat_0.10: PASS (301.56 s)
 (2/2) Host_RHEL.m8.u1.raw.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.rebase_onto_no_backing_file.compat_level.compat_default: PASS (302.21 s)
```

**Comment by hellohellenmao** [comment]
Compatibility test for raw - Pass!

 (01/11) Host_RHEL.m8.u1.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.unsafe_rebase_to_none_existing_backing_file.compat_0.10: PASS (3.59 s)
 (02/11) Host_RHEL.m8.u1.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.unsafe_rebase_to_none_existing_backing_file.compat_default: PASS (3.50 s)
 (03/11) Host_RHEL.m8.u1.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.cache_mode.writethrough: PASS (148.15 s)
 (04/11) Host_RHEL.m8.u1.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.cache_mode.writeback: PASS (146.33 s)
 (05/11) Host_RHEL.m8.u1.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.cache_mode.none: PASS (142.88 s)
 (06/11) Host_RHEL.m8.u1.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.cache_mode.unsafe: PASS (139.95 s)
 (07/11) Host_RHEL.m8.u1.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.cache_mode.directsync: PASS (154.56 s)
 (08/11) Host_RHEL.m8.u1.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.compat_level.compat_0.10: PASS (152.78 s)
 (09/11) Host_RHEL.m8.u1.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.compat_level.compat_default: PASS (146.70 s)
 (10/11) Host_RHEL.m8.u1.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.rebase_onto_no_backing_file.compat_level.compat_0.10: PASS (227.87 s)
 (11/11) Host_RHEL.m8.u1.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.rebase_onto_no_backing_file.compat_level.compat_default: PASS (251.38 s)

**Comment by lolyu** [comment]
@luckyh please help to review, thanks.

**Comment by hellohellenmao** [comment]
Compatibility test for qcow2 - Pass!

Note:
For the last failed case, it is a known issue which is under fix. 

 (01/16) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_img_negative.rebase.nonexist_old_backing_file: CANCEL: Got host qemu version:4.1.0-13, which is not in [, 4.1.0) (3.42 s)
 (02/16) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_img_negative.rebase.nonexist_new_backing_file: PASS (3.66 s)
 (03/16) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snB.to_base: PASS (552.00 s)
 (04/16) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snB.to_base.cache_mode.none: PASS (422.66 s)
 (05/16) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snB.to_base.cache_mode.writeback: PASS (411.84 s)
 (06/16) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snB.to_base.cache_mode.unsafe: PASS (398.64 s)
 (07/16) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snB.to_base.cache_mode.directsync: PASS (413.93 s)
 (08/16) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snB.to_base.cache_mode.writethrough: PASS (409.51 s)
 (09/16) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snB.to_null: PASS (495.53 s)
 (10/16) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snC.to_base: PASS (535.89 s)
 (11/16) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snC.to_new: PASS (581.30 s)
 (12/16) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snC.to_new.restore_to_snB: PASS (701.01 s)
 (13/16) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snD.to_snB_snA_base: PASS (889.36 s)
 (14/16) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_img.rebase.unsafe: PASS (62.86 s)
 (15/16) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_img.rebase.show_progress.show_progress_on: PASS (62.04 s)
 (16/16) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_img.rebase.show_progress.show_progress_off: FAIL: Fail to get expected result!Output: b'', expected pattern: \(\d+\.\d+/100%\) (4.18 s)

File: virttest/qemu_storage.py [hunk 1]
```diff
@@ -439,28 +441,38 @@ def rebase(self, params, cache_mode=None):
...
+            cmd_dict["secret_objects"] = " ".join(secret_objects)
+        if self.encryption_config.key_secret:
+            cmd_dict["image_filename"] = "'%s'" % get_image_json(
+                self.tag, self.params, self.root_dir)
+            cmd_dict.pop("image_format")
         if self.base_tag:
             if self.base_tag == "null":
-                cmd += " -b \"\" -F %s %s" % (self.base_format,
-                                              self.image_filename)
+                cmd_dict["backing_file"] = "\"\""
```
Comment (luckyh) [inline virttest/qemu_storage.py]: feeling dizzy, would it be `'""'` or `"''"`?

**Comment by luckyh** [comment]
Thanks all.

**Pull request merged**